collagraph/cgx/cgx.py

```diff
@@ -17,6 +18,8 @@
 DIRECTIVE_ON = f"{DIRECTIVE_PREFIX}on"
 FOR_LOOP_OUTPUT = "for_loop_output"
 
+COMPONENT_CLASS_DEFINITION = re.compile(r"class\s*(.*?)\s*\(.*Component\s*\)\s*:")
```

Comment:
does this regex allow for mixins etc? I imagine that might be something users would want to be able to do. example:

```python
class Ticker(cg.Component, MyUsefulMixin):
    ....
```

Reply:
Actually, I've removed the check for Component, since you might want to subclass another component, so I can't check for the string 'Component'. Instead I'll just look for any class definitions, loop over them and see if there is but one that is a subclass of Component.